### PR TITLE
Adding RecoveryServices module

### PR DESCRIPTION
**Pull request by dragonfly91**
Corresponding Swagger PR (merged): https://github.com/Azure/azure-rest-api-specs/pull/552

### CODE REVIEW

**Comment by markcowl** [comment]
Hold review until rest api spec is merged

**Comment by dragonfly91** [comment]
The REST API spec is now merged. Can you please start the review?

**Comment by shahabhijeet** [comment]
@dragonfly91 It seems you have added new functionality, can you bump up the minor version

File: src/ResourceManagement/RecoveryServices/Microsoft.Azure.Management.RecoveryServices/generate.cmd [hunk 1]
```diff
@@ -0,0 +1,17 @@
+::
+:: Microsoft Azure SDK for Net - Generate library code
+:: Copyright (C) Microsoft Corporation. All Rights Reserved.
+::
+
+@echo off
+set autoRestVersion=0.16.0-Nightly20160406
```
Comment (cormacpayne) [inline src/ResourceManagement/RecoveryServices/Microsoft.Azure.Management.RecoveryServices/generate.cmd]: @dragonfly91 please use the latest version of the generator

File: src/ResourceManagement/RecoveryServices/Microsoft.Azure.Management.RecoveryServices/project.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
+  "version": "4.0.1-preview",
```
Comment (cormacpayne) [inline src/ResourceManagement/RecoveryServices/Microsoft.Azure.Management.RecoveryServices/project.json]: @dragonfly91 if the current version is 3.0.1-preview, why is it going to 4.0.1-preview?

Comment (dragonfly91) [inline src/ResourceManagement/RecoveryServices/Microsoft.Azure.Management.RecoveryServices/project.json]: @cormacpayne Moving to Autorest from Hyak has caused some breaking changes in the client, thus I've increased the major version.

**Comment by cormacpayne** [comment]
LGTM

**Pull request merged**